providers/default.rb

```diff
@@ -95,115 +80,82 @@ def get_java_path(resource)
       "================================================================\n"
   end
 
-  converge_by("Create upstart script for \"#{app_name}\" in /etc/init") do
-
-    # Upstart script created based on the app_name
-    t = template "/etc/init/#{app_name}.conf" do
-      source new_resource.init_script_source
-      cookbook new_resource.init_script_cookbook
-
-      mode 0644
-      owner 'root'
-      group 'root'
-      variables(
-        app_name: app_name,
-        java_bin: get_java_path(new_resource),
-        app_path: app_path,
-        app_user: app_user,
-        pid_file: pid_file,
-        jar_file: jar_file,
-        jvm_options: new_resource.jvm_options,
-        arguments: new_resource.arguments
-      )
-      notifies :restart, "service[#{app_name}]" if jar_exists
-    end
+  # put application arguments into an array in the correct order:
+  # -jar, jar file, all provided arguments
+  # optionally prefixed by jvm_options, if provided.
+  app_args = ['-jar', jar_file, *new_resource.arguments.split(' ')]
+  unless new_resource.jvm_options.empty?
+    app_args = new_resource.jvm_options.split(' ').push(*app_args)
+  end
 
-    updated_notification = (updated_notification || t.updated_by_last_action?)
+  # Upstart script created based on the app_name
+  pr = pleaserun app_name do
+    name app_name
+    program get_java_path(new_resource)
+    args app_args
+    user app_user
+    action :create
+    notifies :restart, "service[#{app_name}]" if jar_exists
+  end
 
-    # Since this is an upstart script, doing a symlink to
-    # 'upstart-job' will work, and include a deprecation notice.
-    if platform?('ubuntu')
-      link "/etc/init.d/#{app_name}" do
-        to '/lib/init/upstart-job'
-        only_if 'test -f /lib/init/upstart-job && test -d /etc/init.d'
-      end
+  # Since this is an upstart script, doing a symlink to
+  # 'upstart-job' will work, and include a deprecation notice.
+  if platform?('ubuntu')
+    link "/etc/init.d/#{app_name}" do
+      to '/lib/init/upstart-job'
+      only_if 'test -f /lib/init/upstart-job && test -d /etc/init.d'
     end
-
   end
 
-  converge_by("Starting service: #{app_name}") do
+  s = service app_name do
+    provider Chef::Provider::Service::Upstart
```

Comment:
Hm. This might be an issue. By default, `pleaserun` will create an upstart init script on Ubuntu 14.04. However, for other distros, it might use something else. I think we can just remove the provider line here (and below)